keras/callbacks.py

```diff
@@ -1801,23 +1801,25 @@ def on_epoch_end(self, epoch, logs=None):
     if self.restore_best_weights and self.best_weights is None:
       # Restore the weights after first epoch if no progress is ever made.
       self.best_weights = self.model.get_weights()
-
-    self.wait += 1
-    if self._is_improvement(current, self.best):
       self.best = current
-      if self.restore_best_weights:
-        self.best_weights = self.model.get_weights()
-      # Only restart wait if we beat both the baseline and our previous best.
-      if self.baseline is None or self._is_improvement(current, self.baseline):
-        self.wait = 0
-
-    if self.wait >= self.patience:
-      self.stopped_epoch = epoch
-      self.model.stop_training = True
-      if self.restore_best_weights and self.best_weights is not None:
-        if self.verbose > 0:
-          print('Restoring model weights from the end of the best epoch.')
-        self.model.set_weights(self.best_weights)
+    else:
+      # Only judge the best after first epoch
+      self.wait += 1
```

Comment:
I really don't see how incrementing `self.wait` at the end of each epoch is the correct behavior.

You have access to the epoch counter `epoch`, if you just want to skip the first epoch you can? E.g. if it's the first epoch, then set the best score / weights and continue.

Reply:
You're right. I've modified it.